macaroon_service.go

```diff
@@ -231,12 +233,53 @@ func (ms *MacaroonService) Start() error {
 
 	// There are situations in which we don't want a macaroon to be created
 	// on disk (for example when running inside LiT stateless integrated
-	// mode). For any other cases, we create macaroon files in the default
-	// directory.
-	if ms.cfg.StatelessInit || lnrpc.FileExists(ms.cfg.MacaroonPath) {
+	// mode).
+	if ms.cfg.StatelessInit {
 		return nil
 	}
 
+	// If there are no required permissions, then we don't have to generate
+	// or re-generate a macaroon and so we can exit here.
+	if len(ms.cfg.RequiredPerms) == 0 {
+		return nil
```

Comment:
I think we should return an error here instead. A macaroon without any permissions doesn't really serve any purpose and is potentially a security risk. So I think it makes sense to move this check to the `NewMacaroonService` function.

Reply:
Hmm so we have another test in the lndclient repo `TestMacaroonServiceMigration` that will fail if we return an error here. Since that test isn't creating any macaroon `RequiredPerms` will be `nil`. 
So if we're not ok with returning `nil` if this if clause is true, I see two other alternatives:
1. Change that test to also create a Macaroon, but that would kind of go against the focus of the test.
2. Instead of matching using the `RequiredPerms`, match by checking the `cfg.MacaroonPath` and checking if the path ends with `.macaroon`, and then return nil. Not really a fan of that alternative either as it will be a bit confusing if people integrate the `lndclient` repo that the repo will return `nil` and not an error if the macaroon path if incorrect.

Which version would you prefer :)?

Reply:
If I interpret the diff correctly, I think `TestMacaroonServiceMigration` did create a macaroon before, just with empty permissions (and it didn't explicitly check for the creation).
So I think option 1 is fine, just adding a dummy permission to that test (1 line change).

Reply:
Ok thanks! I added generation of a macaroon to the `TestMacaroonServiceMigration` test :)! The previous diff succeeded without the macaroon as the code would exit early since the `tempDirPath` existed and `lnrpc.FileExists(ms.cfg.MacaroonPath)` would then match.